storage/src/main/java/org/apache/kafka/storage/internals/log/UnifiedLog.java

```diff
@@ -647,8 +647,9 @@ private LogOffsetMetadata fetchLastStableOffsetMetadata() throws IOException {
 
         // cache the current high watermark to avoid a concurrent update invalidating the range check
         LogOffsetMetadata highWatermarkMetadata = fetchHighWatermarkMetadata();
-        if (firstUnstableOffsetMetadata.isPresent() && firstUnstableOffsetMetadata.get().messageOffset < highWatermarkMetadata.messageOffset) {
-            LogOffsetMetadata lom = firstUnstableOffsetMetadata.get();
+        Optional<LogOffsetMetadata> firstUnstableOffsetMetadataCopy = firstUnstableOffsetMetadata;
```

Comment:
Should we add a comment or update the comment just above?

Reply:
Done